README.md

```diff
@@ -44,6 +44,20 @@ class BaseObject < GraphQL::Schema::Object
 end
 ```
 
+Include the `ApolloFederation::Interface` module in your base interface module:
+
+```ruby
+module BaseInterface
+  include GraphQL::Schema::Interface
+
+  field_class BaseField
+
+  definition_methods do
```

Comment:
This interface seems a little clunky and inconsistent. We could make it so you simply `include ApolloFederation::Interface` in the module definition by adding the definition methods in an `included` hook:
```ruby
module ApolloFederation
  module Interface
    def self.included(klass)
      klass.definition_methods do
        include DefinitionMethods
      end
    end

    module DefinitionMethods
      include HasDirectives

      def extend_type
        add_directive(name: 'extends')
      end

      def key(fields:)
        add_directive(
          name: 'key',
          arguments: [
            name: 'fields',
            values: fields,
          ],
        )
      end
    end
  end
end
```

What do you think?

Reply:
Ok makes sense, I'll see if I can make the change.

Reply:
@rylanc done

Reply:
Aaand linted 😁